boa/src/builtins/array/mod.rs

```diff
@@ -1782,6 +1783,183 @@ impl Array {
+        let arr = Self::array_species_create(&o, actual_delete_count, context)?;
+        // 12. Let k be 0.
+        // 13. Repeat, while k < actualDeleteCount,
+        for k in 0..actual_delete_count {
+            // a. Let from be ! ToString(𝔽(actualStart + k)).
+            // b. Let fromPresent be ? HasProperty(O, from).
+            let from_present = o.has_property(actual_start + k, context)?;
+            // c. If fromPresent is true, then
+            if from_present {
+                // i. Let fromValue be ? Get(O, from).
+                let from_value = o.get(actual_start + k, context)?;
+                // ii. Perform ? CreateDataPropertyOrThrow(A, ! ToString(𝔽(k)), fromValue).
+                arr.create_data_property_or_throw(k, from_value, context)?;
+            }
+            // d. Set k to k + 1.
+        }
+
+        // 14. Perform ? Set(A, "length", 𝔽(actualDeleteCount), true).
+        arr.set("length", actual_delete_count, true, context)?;
+
+        // 15. Let itemCount be the number of elements in items.
+        let item_count = items.len();
+
+        match item_count.cmp(&actual_delete_count) {
+            // 16. If itemCount < actualDeleteCount, then
+            Ordering::Less => {
+                //     a. Set k to actualStart.
+                //     b. Repeat, while k < (len - actualDeleteCount),
+                for k in actual_start..(len - actual_delete_count) {
+                    // i. Let from be ! ToString(𝔽(k + actualDeleteCount)).
+                    let from = k + actual_delete_count;
+                    // ii. Let to be ! ToString(𝔽(k + itemCount)).
+                    let to = k + item_count;
+                    // iii. Let fromPresent be ? HasProperty(O, from).
+                    let from_present = o.has_property(from, context)?;
+                    // iv. If fromPresent is true, then
+                    if from_present {
+                        // 1. Let fromValue be ? Get(O, from).
+                        let from_value = o.get(from, context)?;
+                        // 2. Perform ? Set(O, to, fromValue, true).
+                        o.set(to, from_value, true, context)?;
+                    // v. Else,
+                    } else {
+                        // 1. Assert: fromPresent is false.
+                        debug_assert!(!from_present);
+                        // 2. Perform ? DeletePropertyOrThrow(O, to).
+                        o.delete_property_or_throw(to, context)?;
+                    }
+                    // vi. Set k to k + 1.
+                }
+                // c. Set k to len.
+                // d. Repeat, while k > (len - actualDeleteCount + itemCount),
+                for k in ((len - actual_delete_count + item_count + 1)..=len).rev() {
+                    // i. Perform ? DeletePropertyOrThrow(O, ! ToString(𝔽(k - 1))).
+                    o.delete_property_or_throw(k - 1, context)?;
+                    // ii. Set k to k - 1.
+                }
```

Comment:
```suggestion
                for k in ((len - actual_delete_count + item_count)..len).rev() {
                    // i. Perform ? DeletePropertyOrThrow(O, ! ToString(𝔽(k - 1))).
                    o.delete_property_or_throw(k, context)?;
                    // ii. Set k to k - 1.
                }
```
I don't know why the spec convolutes so much the index. Can you confirm if this change works?

Reply:
Maybe we should open a PR in the spec itself, this doesn't make much sense

Reply:
Same conformance but I'm not sure why the spec specifically does this.

Reply:
Then change both to this. I don't see why they sum 1 to everything just to subtract one inside the loop.